src/rand.jl

```diff
@@ -1,3 +1,14 @@
+using Random
+
 Base.rand(X::Interval{T}) where {T} = X.lo + rand(T) * (X.hi - X.lo)
 
 Base.rand(X::IntervalBox) = rand.(X)
+
+
+Base.eltype(::Type{Interval{Float64}}) = Float64
```

Comment:
Unfortunately this line disagrees with https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/263.

Is this really necessary for the random samplers? That seems strange. (But it could be.)

Reply:
That line is also used to change the default return type of the random samplers according to the docs, I used it to convert the return type of the function from Interval{Float64} to a Float Vector.

Reply:
Yes I see. What I'm saying is that that conflicts with the way that we need to define `eltype` for `Interval`s in order for broadcasting to work. Maybe you can ask about this on Slack or raise it as an issue on the JuliaLang issue tracker.

Reply:
I think the solution is that the `Random.eltype` function should be distinguished from `Base.eltype`. I'm not sure this is possible right now?

Reply:
I reckon we can alter the return type of broadcasting in the `Base.similar` function.
Check [https://docs.julialang.org/en/v1/manual/interfaces/#Selecting-an-appropriate-output-array-1](url)

Reply:
After asking on Slack it seems that the following solves the issue:
```
julia> Random.gentype(::Type{Interval{T}}) where {T} = T
```

Reply:
Yes it does.